### PR TITLE
[GEOT-7633] Move ZoomContext/ZoomContextFinder class set from YSLD to main

### PR DESCRIPTION
**Pull request by aaime**
[GEOT-7633](https://osgeo-org.atlassian.net/browse/GEOT-7633) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geotools&utm_campaign=badge_info)

As per list discussion, moving the ZoomContext related classes from YSLD to main.
The old classes in YSLD use the "extend and deprecate" approach.
@smithkm @jodygarnett 

Downstream build failure is to be expected, the following PR adapts GeoServer to the API change:
https://github.com/geoserver/geoserver/pull/7838

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geotools/geotools/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geotools.org/latest/developer/procedures/contribution_license.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [ ] Avoid [Java 9+ split packages](http://tutorials.jenkov.com/java/modules.html#split-packages-not-allowed).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [x] [Documentation](https://github.com/geotools/geotools/tree/main/docs) has been updated (if change is visible to end users).
- [x] There is an issue in [GeoTools Jira](https://osgeo-org.atlassian.net/projects/GEOT) (except for changes not visible to end users). 
- [x] Commit message(s) must be in the form ``[GEOT-XYZW] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] The commit targets a single objective (if multiple focuses cannot be avoided, each one is in its own commit, and has a separate ticket describing it).

### CODE REVIEW

File: docs/user/library/main/zoom.rst [hunk 1]
```diff
@@ -0,0 +1,22 @@
+Zoom to Scale denominators
+==========================
+
+The classes found in the ``org.geotools.styling.zoom`` package provide a way to convert zoom levels into scale denominators. In particular:
+ 
+ * ``ZoomContextFactory`` is a factory class that provides a way to create a ``ZoomContext`` object for a given Tile Matrix Set, by name.
+ * ``ZoomContext`` is an interface that provides a way to convert zoom levels into scale denominators.
+ * ``ScalerRange`` is an interface that describe a range of scale denominators.
+
+The ``WellKnownZoomContextFactory`` class provides way to create a ``ZoomContext`` object for well
```
Comment (jodygarnett) [inline docs/user/library/main/zoom.rst:10]:
```suggestion
The ``WellKnownZoomContextFactory`` class provides way to create a ``ZoomContext`` object for well-known
```

File: modules/library/main/src/test/java/org/geotools/styling/zoom/TestUtils.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.describedAs;
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.lessThanOrEqualTo;
+
+import org.hamcrest.Matcher;
+import org.hamcrest.Matchers;
+
+public enum TestUtils {
+    ;
```
Comment (jodygarnett) [inline modules/library/main/src/test/java/org/geotools/styling/zoom/TestUtils.java:29]:
```suggestion
```

File: modules/library/main/src/test/java/org/geotools/styling/zoom/TestUtils.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.describedAs;
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.lessThanOrEqualTo;
+
+import org.hamcrest.Matcher;
+import org.hamcrest.Matchers;
+
+public enum TestUtils {
+
```
Comment (jodygarnett) [inline modules/library/main/src/test/java/org/geotools/styling/zoom/TestUtils.java:29]:
```suggestion
    ;
```

File: modules/library/main/src/test/java/org/geotools/styling/zoom/TestUtils.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.describedAs;
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.lessThanOrEqualTo;
+
+import org.hamcrest.Matcher;
+import org.hamcrest.Matchers;
+
+public enum TestUtils {
+    ;
```
Comment (jodygarnett) [inline modules/library/main/src/test/java/org/geotools/styling/zoom/TestUtils.java:29]:
```suggestion
    ;
    
```

File: modules/library/main/src/test/java/org/geotools/styling/zoom/TestUtils.java (hunk 2 above)
Comment (aaime) [inline modules/library/main/src/test/java/org/geotools/styling/zoom/TestUtils.java:29]: Weird eh? Necessary semicolumn for an empty enum. I believe this was @smithkm attempt to ensure this class could not be instantiated.

**Pull request merged**